### PR TITLE
add .destroy to noAutoDestroy iterator

### LINKED ISSUE #94: Add local `destroy` in `MachineRunnerIterableIterator`
`MachineRunnerIterableIterator` is an object that's incorporated in `machine` to provide `next`, `peek`, and `destroy`, but also returned by the `noAutoDestroy` method.

The `noAutoDestroy` variant, unlike the original `machine` it comes from, doesn't have a `destroy` method. Users have to resort to `breaking/throwing` from the `for-await` loop to destroy it.

With local `destroy` introduced, users can directly destroy a `noAutoDestroy` copy without breaking from a `for-await` loop

### PR DESCRIPTION
**Pull request by Kelerchian**
Resolves #94

### CODE REVIEW

File: machine-runner/tests/esm/runner.test.ts [hunk 1]
```diff
@@ -801,27 +801,38 @@ describe('machine as async generator', () => {
...
 
       // peek should come before next
-      const cres2 = await cloned.peek()
-      const mres2 = await machine.next()
-
-      expect(mres2.done).toBeFalsy()
-      expect(cres2.done).toBeTruthy()
+      const cres1_2 = await cloned1.peek()
+      const cres2_2 = await cloned1.peek()
+      const cres3_2 = await cloned1.peek()
```
Comment (rkuhn) [inline machine-runner/tests/esm/runner.test.ts:829]: isn’t the point of `.noAutoDestroy()` (since my changes) that this should also work with `.next()`?

**Review by rkuhn — COMMENTED** [review]
I don’t have all of it in active working memory: didn’t you mention that the `.noAutoDestroy()` result had the wrong destruction wired in somehow?

File: machine-runner/tests/esm/runner.test.ts (hunk 1 above)
Comment (Kelerchian) [reply machine-runner/tests/esm/runner.test.ts:829]: ah, my bad. those should be cloned1, cloned2, and cloned 3

**Comment by Kelerchian** [comment]
> didn’t you mention that the .noAutoDestroy() result had the wrong destruction wired in somehow?

No. The wrong wiring was a bug in user's code.
However, that bug reminds me that missing pieces such as this can easily cause (although indirectly) a semantic misunderstanding.

**Pull request merged**